### ISSUE TITLE
`onFileLoading` Callback function isn't reset between picking files and directory

### ISSUE THREAD

**Issue by BernardinD** [miguelpruivo/flutter_file_picker#863]
Before creating an issue, make sure that you are on the latest `file_picker` version and that there aren't already any similar opened inssues. Also, check if it isn't described on the [Wiki](https://github.com/miguelpruivo/flutter_file_picker/wiki), specially on [Troubleshooting](https://github.com/miguelpruivo/flutter_file_picker/wiki/Troubleshooting) page.

Also, sometimes a simple `flutter clean` and `flutter build` again with latest file_picker version, may end up by fixing cached issues, so I encourage you to first do so.

**Describe the bug**
I am using both `FilePicker.platform.getDirectoryPath` and `FilePicker.platform.pickFiles` in an app and the `onFileLoading:` callback for picking files is also being run for picking directories and doesn't go away.

**Platform**
- [x] Android
- [] iOS
- [] Web
- [] Desktop (Go)

**Platform OS version**
4.2.0

**How are you picking?**
```
_directoryPath =  await FilePicker.platform.getDirectoryPath();
```

**Details to reproduce the issue**
```
Future foo1() async{
  paths = (await FilePicker.platform.pickFiles(
          type: _pickingType,
          allowMultiple: _multiPick,
          onFileLoading: (status) {
            debugPrint("filepicker status: $status");
            if(status == FilePickerStatus.picking) {
              debugPrint("picking...");
              timer.start();
            }
            else if(status == FilePickerStatus.done){
              debugPrint("done...");
              debugPrint("Elasped (picking): ${timer.elapsedMilliseconds}ms");
              timer.stop();
            }
          }))
          ?.files;
}

Future foo2() async{
  _directoryPath =  await FilePicker.platform.getDirectoryPath();
}
```

**Flutter Version details**
```
[√] Flutter (Channel unknown, 2.2.3, on Microsoft Windows [Version 10.0.19042.1288], locale en-US)
    • Flutter version 2.2.3 at C:\src\flutter
    • Framework revision f4abaa0735 (4 months ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\deziu\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.4)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30517.126
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Community Edition (version 2020.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.2.2
    • Flutter plugin version 49.0.4
    • Dart plugin version 202.7319.5

[√] Connected device (4 available)
    • SM G970U (mobile) • R38MB0BMZ3E • android-arm64  • Android 11 (API 30)
    • Windows (desktop) • windows     • windows-x64    • Microsoft Windows [Version 10.0.19042.1288]
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 95.0.4638.69
    • Edge (web)        • edge        • web-javascript • Microsoft Edge 94.0.992.31

• No issues found!
```

**Comment by BernardinD** [comment]
I've also tried overriding the callback right before picking the directory using:

```
// Reset callback function
    try {
      await FilePicker.platform.pickFiles(
          type: FileType.image,
          onFileLoading: (_) => debugPrint(""), allowedExtensions: ["fail"]);
    }
    catch (e){debugPrint("changeDir >> $e");}
```

But the callback for picking files is still called

**Comment by miguelpruivo** [comment]
Fixed with 4.2.1. Thank you for reporting. 🎉
